operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -118,9 +150,58 @@ func (c *cesManager) RemoveCEPMapping(cep *cilium_v2.CoreCiliumEndpoint, ns stri
 		if c.mapping.countCEPsInCES(cesName) == 0 {
 			c.mapping.deleteCES(cesName)
 		}
-		return NewCESKey(cesName.string(), ns)
+		return []CESKey{NewCESKey(cesName.string(), pod.Namespace)}
+	}
+	return []CESKey{CESKey(resource.Key{})}
+}
+
+func (c *cesManager) UpdateNodeMapping(node *cilium_v2.CiliumNode) []CESKey {
+	newKey := NewEncryptionKey(node.Spec.Encryption.Key)
+	name := NewNodeName(node.Name)
+	if c.mapping.hasNode(name) {
+		// If node encryption key is changed, update affected CES.
+		oldKey := c.mapping.getNodeEncryptionKey(name)
+		if oldKey != newKey {
+			return c.mapping.insertNode(name, newKey)
+		}
+	} else {
+		return c.mapping.insertNode(name, newKey)
+	}
+	return nil
+}
+
+func (c *cesManager) RemoveNodeMapping(node *cilium_v2.CiliumNode) []CESKey {
+	if c.mapping.hasNode(NewNodeName(node.Name)) {
+		return c.mapping.deleteNode(NewNodeName(node.Name))
+	}
+	return nil
+}
+
+func (c *cesManager) UpdateIdentityMapping(id *cilium_v2.CiliumIdentity) []CESKey {
+	cidName, gidLabels := cidToGidLabels(id)
+	return c.mapping.insertCID(cidName, gidLabels)
+}
+
+func (c *cesManager) RemoveIdentityMapping(id *cilium_v2.CiliumIdentity) []CESKey {
+	cidName, gidLabels := cidToGidLabels(id)
+	if c.mapping.hasCID(cidName, gidLabels) {
+		return c.mapping.deleteCID(cidName, gidLabels)
 	}
-	return CESKey(resource.Key{})
+	return nil
+}
+
+func (c *cesManager) UpdateNamespaceMapping(ns *slim_corev1.Namespace) []CESKey {
+	return c.mapping.getCESInNs(ns.GetName())
+}
```

Comment:
The name of this function is misleading, as no update appears to be actually performed.